From d4c3133378b377ee519ea50247339cd61221fc47 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 21 Jun 2023 09:20:36 +0100
Subject: [PATCH] overlays: allo-katana-dac-audio: Reduce I2C clock

Higher speeds have been shown to cause data corruption on a Pi 4,
possibly due to clock-stretching.

See: https://github.com/raspberrypi/linux/issues/5511

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/allo-katana-dac-audio-overlay.dts | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/arm/boot/dts/overlays/allo-katana-dac-audio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/allo-katana-dac-audio-overlay.dts
@@ -30,6 +30,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			status = "okay";
+			clock-frequency = <50000>;
 
 			allo-katana-codec@30 {
 				#sound-dai-cells = <0>;
